lib/faraday_middleware/response/follow_redirects.rb

```diff
@@ -102,6 +102,8 @@ def update_env(env, request_body, response)
       redirect_to_url = safe_escape(response['location'] || '')
       env[:url] += redirect_to_url
 
+      ENV_TO_CLEAR.each { |key| env.delete key }
+
```

Comment:
This needs to happen before we access env[:body] otherwise it returns the response_body and not the request_body